lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -555,38 +556,58 @@ noExtraTxBodyContent = ExtraTxBodyContent [] [] [] id
 -- To avoid the need for `ledger -> wallet` conversions, this function can only
 -- be used to *add* tx body content.
 updateSealedTx
-    :: ExtraTxBodyContent
+    :: TxUpdate
     -> SealedTx
     -> Either ErrUpdateSealedTx SealedTx
 updateSealedTx extraContent (cardanoTx -> InAnyCardanoEra _era tx) = do
 
     -- NOTE: The script witnesses are carried along with the cardano-api
     -- `anyEraBody`.
     let (Cardano.Tx anyEraBody existingKeyWits) = tx
-    body' <- modifyLedgerTxBody extraContent anyEraBody
+    body' <- modifyLedgerTx extraContent anyEraBody
 
     unless (null existingKeyWits) $
        Left $ ErrExistingKeyWitnesses $ length existingKeyWits
 
     return $ sealedTxFromCardanoBody body'
 
   where
-    modifyLedgerTxBody
-        :: ExtraTxBodyContent
+    modifyLedgerTx
+        :: forall era. ()
+        => TxUpdate
         -> Cardano.TxBody era
         -> Either ErrUpdateSealedTx (Cardano.TxBody era)
-    modifyLedgerTxBody ebc (Cardano.ShelleyTxBody shelleyEra bod scripts scriptData aux val) =
-            Right $ Cardano.ShelleyTxBody shelleyEra (adjust ebc shelleyEra bod) scripts scriptData aux val
+    modifyLedgerTx ebc (Cardano.ShelleyTxBody shelleyEra bod scripts scriptData aux val) =
+            Right $ Cardano.ShelleyTxBody shelleyEra
+                (adjustBody ebc shelleyEra bod)
+                scripts
+                (modifyRedeemers scriptData)
+                aux
+                val
       where
+        modifyRedeemers :: Cardano.TxBodyScriptData era -> Cardano.TxBodyScriptData era
```

Comment:
:+1:

Reply:
maybe it is absolutely obvious `modifyRedeemers` is correct, but feel some property test showing the case would not be harmful;-)

Reply:
What about updating this property https://github.com/input-output-hk/cardano-wallet/blob/master/lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs#L1749 ?